### PR TITLE
show gas estimation for constructor

### LINKED ISSUE #233: gas estimetion/selection doesn't work for constructor body

### PR DESCRIPTION
**Pull request by LianaHus**
fixes https://github.com/ethereum/mix/issues/233
- style fixes
- gas estimation for constructor

### CODE REVIEW

File: src/ClientModel.cpp [hunk 1]
```diff
@@ -1251,8 +1280,27 @@ void ClientModel::onNewTransaction(RecordLogEntry::TxSource _source)
...
+			gasRequired,
+			gasRefunded,
+			sender,
+			label,
+			inputParameters,
+			returnParameters,
+			logs,
+			_source,
+			exception
+			);
```
Comment (chriseth) [inline src/ClientModel.cpp]: Indentation

**Comment by LianaHus** [comment]
@chriseth sorry what exactly can I move out?

**Comment by LianaHus** [comment]
we always have braces for constructor

File: src/CodeModel.cpp [hunk 2]
```diff
@@ -496,6 +506,21 @@ GasMapWrapper* CodeModel::gasEstimation(solidity::CompilerStack const& _cs)
 				GasMeter::GasConsumption cost = gasItem->second;
 				gasCostsMaps->push(sourceName, itemLocation.start, itemLocation.end, gasToString(cost), cost.isInfinite, GasMap::type::Statement);
 			}
+			// Structural gas costs for constructor
+			if (contractDefinition.constructor())
+			{
+				for (auto gasItem = constructorGasCosts.begin(); gasItem != constructorGasCosts.end(); ++gasItem)
+				{
+					SourceLocation const& itemLocation = gasItem->first->location();
+						SourceLocation const& constructorLocation = contractDefinition.constructor()->location();
```
Comment (chriseth) [inline src/CodeModel.cpp]: This can be moved to before the for loop.

File: src/CodeModel.cpp [hunk 3]
```diff
@@ -496,6 +506,21 @@ GasMapWrapper* CodeModel::gasEstimation(solidity::CompilerStack const& _cs)
...
+						// check whether the location of the item is in constructor scope
+						if (itemLocation.start >= constructorLocation.start && itemLocation.end <= constructorLocation.end)
```
Comment (chriseth) [inline src/CodeModel.cpp]: Indentation is wrong.

**Pull request merged**